SignalServiceKit/src/Messages/OWSBatchMessageProcessor.m

```diff
@@ -283,6 +287,13 @@ - (void)appIsReady
     [self drainQueue];
 }
 
+- (void)yapDatabaseModified:(NSNotification *)notification
+{
+    OWSAssertIsOnMainThread();
+
+    [self drainQueue];
```

Comment:
The core fix - we used to kick off queue processing in `enqueueEnvelopeData`, but this won't work any more (see the comment below).  So now we observe changes to the database.